src/simple.rs

```diff
@@ -46,7 +46,59 @@ impl SimpleClient {
         .await
     }
 
+    /// Execute redis command and stream response
+    pub fn stream<U>(
+        self,
+        cmd: U,
```

Comment:
are there any other commands for streaming except "subscribe"?

Reply:
only subset of "subscribe" command do real streaming (pushing data to client)

Reply:
then should `stream()` accept configuration for subscribe instead of command? there is no point to accept any command when it can accept only one cmd

Reply:
after subscribe command client can send subscribe again or unsubscribe for one or all channels to turn connection to regular request-response mode.
please check out  new implementation of `SimpleClient` and `SimpleClient::stream`

Reply:
1. unsubscribe could be executed on `drop`
2. I think simple client should be consumed by `stream` call, otherwise it could break stream. subscriptions management could be part of returned object (OutputStream)
3. turning stream back to simple client could be part of OutputStream api
4. I think name for OutputStream should be different

Reply:
> 1. unsubscribe could be executed on `drop`
it's not necessary to send unsubscribe on connection object `drop` and if send it on `drop` stream object cause messages from redis about all unsubscribed channels if they was.

Unsubscribe  command must be called in explecit way by lib user

> 2. I think simple client should be consumed by `stream` call, otherwise it could break stream. subscriptions management could be part of returned object (OutputStream)
> 3. turning stream back to simple client could be part of OutputStream api
> 4. I think name for OutputStream should be different
thanks. i will try to do this tasks some time later.  it's not easy for me

curretly cleanup subscription message parsing and add unsbuscribe command

sorry for by poor English

Reply:
> 1. unsubscribe could be executed on `drop`
see comment above - unsubscribe must be called in explicit way by lib users

> 2. I think simple client should be consumed by `stream` call, otherwise it could break stream. subscriptions management could be part of returned object (OutputStream)
done, not it `SimplClient::subscribe(self, cmd) -> Result<SubscriptionClient, ..>`
> 3. turning stream back to simple client could be part of OutputStream api
done, `SubscriptionClient::into_client(self) -> SimpleClient` 
But connection may stay in "push mode"
I don't know how to reset this state of connection back to request-response without artefacts

> 4. I think name for OutputStream should be different
done. Now it `SubscriptionClient`

Reply:
@fafhrd91 please take a look again as you can

1) (p|s)subscribe and (p|s)unsubscribe support, plus `RESET` command for `SimpleClient`

2) Any reasons to implement subscription behavior on `Client`?

Reply:
looks good.